go/admin_getbucketsessionid.go

```diff
@@ -3,13 +3,16 @@ package bucketclient
 import (
 	"context"
 	"fmt"
+	"net/url"
 	"strconv"
 )
 
 // AdminGetBucketSessionID returns the raft session ID of the given bucket.
-// Returns -1 and an error if the bucket doesn't exist, or if a request error occurs.
+// Returns 0 and an error if the bucket doesn't exist, or if a request error occurs.
 func (client *BucketClient) AdminGetBucketSessionID(ctx context.Context, bucketName string) (int, error) {
-	resource := fmt.Sprintf("/_/buckets/%s/id", bucketName)
+	// Escape the bucket name to avoid any risk to inadvertently or maliciously
+	// call another route with an incorrect/crafted bucket name containing slashes.
+	resource := fmt.Sprintf("/_/buckets/%s/id", url.PathEscape(bucketName))
 	responseBody, err := client.Request(ctx, "AdminGetBucketSessionID", "GET", resource)
 	if err != nil {
 		return 0, err
```

Comment:
Not related to the changes in this PR, but should we log the error before returning a 0?
May be there are other ways we get to know what happened.

Reply:
The error object of type `BucketClientError` should contain the most relevant info about the error. And I didn't add logging support in this module, as I felt logging was better handled by the calling application.